[Bug]: build of 23.1.5 with gcc 13 fails with overloaded-virtual

### Which component impacted?

Build

### Is it regression? Good in old configuration?

None

### What happened?

Build media-driver with gcc-13-20230402.
builds fine with gcc-12.2

### What's the usage scenario when you are seeing the problem?

Others

### What impacted?

Build fails using gcc-13 20230402
```
[155/1453] Building CXX object media_driver/CMakeFiles/iHD_drv_video_COMMON.dir/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_g12_X.cpp.o
FAILED: 
…
media_driver/CMakeFiles/iHD_drv_video_COMMON.dir/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_g12_X.cpp.o -MF media_driver/CMakeFiles/iHD_drv_video_COMMON.dir/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_g12_X.cpp.o.d -o media_driver/CMakeFiles/iHD_drv_video_COMMON.dir/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_g12_X.cpp.o -c /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-12.0-devel/build/media-driver-23.1.5/media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_g12_X.cpp
In file included from ../media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_generic.h:31,
                 from ../media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_g12_X.h:30,
                 from ../media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_g12_X.cpp:26:
../media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_generic.h: In instantiation of 'class MhwVdboxAvpInterfaceGeneric<mhw_vdbox_avp_g12_X>':
../media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_g12_X.h:38:40:   required from here
../media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_interface.h:779:24: error: 'virtual MOS_STATUS MhwVdboxAvpInterface::AddAvpBsdObjectCmd(PMOS_COMMAND_BUFFER, PMHW_BATCH_BUFFER, MhwVdboxAvpBsdParams*)' was hidden [-Werror=overloaded-virtual=]
  779 |     virtual MOS_STATUS AddAvpBsdObjectCmd(
      |                        ^~~~~~~~~~~~~~~~~~
../media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_generic.h:99:16: note:   by 'MOS_STATUS MhwVdboxAvpInterfaceGeneric<TAvpCmds>::AddAvpBsdObjectCmd(PMOS_COMMAND_BUFFER, MhwVdboxAvpBsdParams*) [with TAvpCmds = mhw_vdbox_avp_g12_X; MOS_STATUS = _MOS_STATUS; PMOS_COMMAND_BUFFER = _MOS_COMMAND_BUFFER*]'
   99 |     MOS_STATUS AddAvpBsdObjectCmd(
      |                ^~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

notes:
- https://gcc.gnu.org/bugzilla/show_bug.cgi?id=20423
- gcc-13 includes -Woverloaded-virtual

### Debug Information

_No response_

### Do you want to contribute a patch to fix the issue?

None

--- a/media_driver/agnostic/Xe_M/Xe_XPM/hw/mhw_vebox_xe_xpm.h
+++ b/media_driver/agnostic/Xe_M/Xe_XPM/hw/mhw_vebox_xe_xpm.h
@@ -110,15 +110,6 @@ public:
         PMOS_COMMAND_BUFFER                 pCmdBuffer,
         PMHW_VEBOX_SURFACE_STATE_CMD_PARAMS pVeboxSurfaceStateCmdParams) override;
 
-    void SetVeboxSurfaces(
-        PMHW_VEBOX_SURFACE_PARAMS                  pSurfaceParam,
-        PMHW_VEBOX_SURFACE_PARAMS                  pDerivedSurfaceParam,
-        PMHW_VEBOX_SURFACE_PARAMS                  pSkinScoreSurfaceParam,
-        mhw_vebox_xe_xpm::VEBOX_SURFACE_STATE_CMD *pVeboxSurfaceState,
-        bool                                       bIsOutputSurface,
-        bool                                       bDIEnable,
-        bool                                       b3DlutEnable);
-
     MOS_STATUS FindVeboxGpuNodeToUse(
         PMHW_VEBOX_GPUNODE_LIMIT pGpuNodeLimit) override;
 
@@ -144,6 +135,18 @@ protected:
     uint32_t m_indexofVebox              = 0;
     uint32_t m_numofVebox                = 1;
     uint32_t m_usingSfc                  = 0;
+   
+    using MhwVeboxInterfaceG12::SetVeboxSurfaces;
+
+    void SetVeboxSurfaces(
+        PMHW_VEBOX_SURFACE_PARAMS                  pSurfaceParam,
+        PMHW_VEBOX_SURFACE_PARAMS                  pDerivedSurfaceParam,
+        PMHW_VEBOX_SURFACE_PARAMS                  pSkinScoreSurfaceParam,
+        mhw_vebox_xe_xpm::VEBOX_SURFACE_STATE_CMD *pVeboxSurfaceState,
+        bool                                       bIsOutputSurface,
+        bool                                       bDIEnable,
+        bool                                       b3DlutEnable);
+
 };
 
 #endif // __MHW_VEBOX_XE_XPM_H__

--- a/media_driver/agnostic/gen12/codec/hal/codechal_encode_csc_ds_mdf_g12.h
+++ b/media_driver/agnostic/gen12/codec/hal/codechal_encode_csc_ds_mdf_g12.h
@@ -202,6 +202,7 @@ protected:
 
      MOS_STATUS CscKernel(
         KernelParams* pParams) override;
+    using CodechalEncodeCscDsG12::InitKernelStateCsc;
     MOS_STATUS InitKernelStateCsc(KernelParams* pParams);
     MOS_STATUS SetupKernelArgsCSC(CmKernel *cmKrnCSCDS, SurfaceParamsCscMdf* surfaceparams);
     MOS_STATUS SetCurbeCscforMDF(CMRT_UMD::vector<uint32_t, 10> & curbeData);

--- a/media_driver/agnostic/gen12/hw/mhw_vebox_g12_X.h
+++ b/media_driver/agnostic/gen12/hw/mhw_vebox_g12_X.h
@@ -336,6 +336,7 @@ private:
     MOS_STATUS ValidateVeboxScalabilityConfig();
 #endif
 
+protected:
     void SetVeboxSurfaces(
         PMHW_VEBOX_SURFACE_PARAMS                 pSurfaceParam,
         PMHW_VEBOX_SURFACE_PARAMS                 pDerivedSurfaceParam,

--- a/media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_generic.h
+++ b/media_driver/agnostic/gen12/hw/vdbox/mhw_vdbox_avp_generic.h
@@ -95,24 +95,5 @@ protected:
 
         return eStatus;
     }
-
-    MOS_STATUS AddAvpBsdObjectCmd(
-        PMOS_COMMAND_BUFFER                 cmdBuffer,
-        MhwVdboxAvpBsdParams                *params)
-    {
-        MOS_STATUS eStatus = MOS_STATUS_SUCCESS;
-
-        MHW_FUNCTION_ENTER;
-        MHW_MI_CHK_NULL(m_osInterface);
-
-        typename TAvpCmds::AVP_BSD_OBJECT_CMD   cmd;
-
-        cmd.DW1.IndirectBsdDataLength       = params->m_bsdDataLength;
-        cmd.DW2.IndirectDataStartAddress    = params->m_bsdDataStartOffset;
-
-        MHW_MI_CHK_STATUS(m_osInterface->pfnAddCommand(cmdBuffer, &cmd, sizeof(cmd)));
-
-        return eStatus;
-    }
 };
 #endif

--- a/media_driver/agnostic/gen9/codec/hal/codechal_encode_avc_g9.h
+++ b/media_driver/agnostic/gen9/codec/hal/codechal_encode_avc_g9.h
@@ -5668,6 +5668,7 @@ public:
     MOS_STATUS GetStatusReport(
         void *status,
         uint16_t numStatus) override;
+    using CodechalEncodeAvcEnc::GetStatusReport;
 
 #if USE_CODECHAL_DEBUG_TOOL
 protected:

--- a/media_driver/agnostic/gen9_skl/codec/hal/codechal_fei_hevc_g9_skl.cpp
+++ b/media_driver/agnostic/gen9_skl/codec/hal/codechal_fei_hevc_g9_skl.cpp
@@ -6410,9 +6410,6 @@ MOS_STATUS CodechalFeiHevcStateG9Skl::FreeEncResources()
 
      m_cmKernelMap.clear();
 
-     CmKernelMapType deallocator;
-     m_cmKernelMap.swap(deallocator);
-
 #endif
 
     return eStatus;

--- a/media_driver/media_softlet/agnostic/Xe_M/Xe_HPM/shared/mediacopy/media_copy_xe_hpm.h
+++ b/media_driver/media_softlet/agnostic/Xe_M/Xe_HPM/shared/mediacopy/media_copy_xe_hpm.h
@@ -50,6 +50,7 @@ public:
     //!
     //! \brief    init function.
     virtual MOS_STATUS Initialize(  PMOS_INTERFACE  osInterface, MhwInterfaces *mhwInterfaces);
+    using MediaCopyBaseState::Initialize;
 
     //!
     //! \brief    render format support.

--- a/media_driver/media_softlet/agnostic/Xe_M/Xe_XPM_base/shared/mediacopy/media_copy_xe_xpm_base.h
+++ b/media_driver/media_softlet/agnostic/Xe_M/Xe_XPM_base/shared/mediacopy/media_copy_xe_xpm_base.h
@@ -49,6 +49,7 @@ public:
     //!
     //! \brief    init function.
     virtual MOS_STATUS Initialize(  PMOS_INTERFACE  osInterface, MhwInterfaces *mhwInterfaces);
+    using MediaCopyBaseState::Initialize;
 
     //!
     //! \brief    surface copy pre process.

--- a/media_driver/media_softlet/agnostic/Xe_M/Xe_XPM_plus/shared/mediacopy/media_copy_xe_xpm_plus.h
+++ b/media_driver/media_softlet/agnostic/Xe_M/Xe_XPM_plus/shared/mediacopy/media_copy_xe_xpm_plus.h
@@ -50,6 +50,7 @@ public:
     //!
     //! \brief    init function.
     virtual MOS_STATUS Initialize(PMOS_INTERFACE osInterface, MhwInterfaces *mhwInterfaces);
+    using MediaCopyBaseState::Initialize;
 
     //!
     //! \brief    render format support.

--- a/media_driver/media_softlet/agnostic/gen12_base/codec/hal/dec/shared/hucitf/decode_huc_packet_creator_g12.h
+++ b/media_driver/media_softlet/agnostic/gen12_base/codec/hal/dec/shared/hucitf/decode_huc_packet_creator_g12.h
@@ -48,6 +48,10 @@ public:
         MediaPipeline       *pipeline,
         MediaTask           *task,
         CodechalHwInterface *hwInterface);
+        
+    using HucPacketCreatorBase::CreateHucCopyPkt;
+    using HucPacketCreatorBase::CreateProbUpdatePkt;
+    using HucPacketCreatorBase::CreateStreamOutInterface;
     MEDIA_CLASS_DEFINE_END(decode__HucPacketCreatorG12)
 };
 

--- a/media_driver/media_softlet/agnostic/m12/m12_0/shared/mediacopy/media_copy_m12_0.h
+++ b/media_driver/media_softlet/agnostic/m12/m12_0/shared/mediacopy/media_copy_m12_0.h
@@ -51,6 +51,7 @@ public:
     //!
     //! \brief    init function.
     virtual MOS_STATUS Initialize(  PMOS_INTERFACE  osInterface, MhwInterfaces *mhwInterfaces);
+    using MediaCopyBaseState::Initialize;
 
     //!
     //! \brief    render format support.

--- a/media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/shared/mediacopy/media_copy_xe_lpm_plus_base.h
+++ b/media_softlet/agnostic/Xe_M_plus/Xe_LPM_plus_base/shared/mediacopy/media_copy_xe_lpm_plus_base.h
@@ -48,6 +48,7 @@ public:
     //!
     //! \brief    init function.
     virtual MOS_STATUS Initialize(PMOS_INTERFACE osInterface, MhwInterfacesNext *mhwInterfaces);
+    using MediaCopyBaseState::Initialize;
 
     //!
     //! \brief    render format support.

--- a/media_softlet/agnostic/common/codec/hal/enc/shared/bufferMgr/encode_tracked_buffer_queue.cpp
+++ b/media_softlet/agnostic/common/codec/hal/enc/shared/bufferMgr/encode_tracked_buffer_queue.cpp
@@ -154,4 +154,9 @@ MOS_STATUS BufferQueue::DestoryResource(void* resource)
     return MOS_STATUS_SUCCESS;
 }
 
+void BufferQueue::SetResourceType(ResourceType resType)
+{ 
+    m_resourceType = resType; 
+}
+
 }

--- a/media_softlet/agnostic/common/codec/hal/enc/shared/bufferMgr/encode_tracked_buffer_queue.h
+++ b/media_softlet/agnostic/common/codec/hal/enc/shared/bufferMgr/encode_tracked_buffer_queue.h
@@ -87,7 +87,7 @@ public:
     //!
     bool SafeToDestory();
 
-    void SetResourceType(ResourceType resType) { m_resourceType = resType; }
+    void SetResourceType(ResourceType resType);
 
 protected:
     //!

--- a/media_softlet/agnostic/common/shared/media_debug_serializer.h
+++ b/media_softlet/agnostic/common/shared/media_debug_serializer.h
@@ -34,6 +34,7 @@
 #include <string>
 #include <typeinfo>
 #include <type_traits>
+#include <stdint.h>
 #include "media_class_trace.h"
 
 template <typename T, typename = void>

--- a/media_softlet/agnostic/common/shared/scalability/media_scalability_multipipe.cpp
+++ b/media_softlet/agnostic/common/shared/scalability/media_scalability_multipipe.cpp
@@ -29,7 +29,7 @@
 
 #include "media_scalability_multipipe.h"
 
-MOS_STATUS MediaScalabilityMultiPipe::UpdateState()
+MOS_STATUS MediaScalabilityMultiPipe::UpdateState(void *statePars)
 {
     return MOS_STATUS_SUCCESS;
 }

--- a/media_softlet/agnostic/common/shared/scalability/media_scalability_multipipe.h
+++ b/media_softlet/agnostic/common/shared/scalability/media_scalability_multipipe.h
@@ -50,7 +50,7 @@ public:
     //! \return MOS_STATUS
     //!         MOS_STATUS_SUCCESS if success, else fail reason
     //!
-    virtual MOS_STATUS UpdateState();
+    virtual MOS_STATUS UpdateState(void *statePars);
 
 protected:
     inline bool IsFirstPipe() { return (m_currentPipe == 0) ? true : false; }

